.github/workflows/legacy.yml

```diff
@@ -0,0 +1,32 @@
+# This workflow validates that (new) variables are not referenced as deprecated legacy variables
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Guard against legacy variables
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ '**' ]
+
+jobs:
+  legacy-validation:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Python 3.11
+      uses: actions/setup-python@v4
+      with:
+        python-version: 3.11
+
+    - name: Install requirements
+      run: pip install -r requirements.txt
```

Comment:
I think you can remove this and just install nomenclature along with pytest below.